### PR TITLE
Add index refresh call after fetch if the list of hits has holes

### PR DESCRIPTION
**Pull request by aleksmaus**
## What does this PR do?

Adds index refresh call after fetch if the list of hits/results has holes.

After discussion with Elasticsearch team it looks like the current global checkpoints
API doesn't not gurantee the list of document without holes on
subsequent index search and there is still a slim chance of this with
our current implementation. Elastisearch team is working on getting the
refresh implemented on thier side, and for now recommended to do the
refresh before fetch if there are holes detected in the result.

Added the check for the holes and refetch, here are the steps:
1. Call Global checkpoints = 5
2. Search = 1, 2, 3, 5.
3. Manual refresh
4. Search and get 4,5
5. Return to step 1

## Why is it important?

Covers the edge case where the fetched documents lists after checkpoint API could still potentially have holes.
In practice we haven't seen this happening, but this change was recommended by Elasticsearch team.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: internal/pkg/monitor/monitor.go [hunk 1]
```diff
@@ -308,6 +368,37 @@ func (m *simpleMonitorT) fetch(ctx context.Context, maxCheckpoint sqn.SeqNo) ([]
 	return hits, nil
 }
 
+// Refreshes index. This is temporary code
+// TODO: Remove this when the refresh is properly imlemented on Eleasticsearch side
```
Comment (scunningham) [inline internal/pkg/monitor/monitor.go:372]: spelling imlemented

**Review by scunningham — APPROVED** [review]
Can you change comment to indicate when we expect ES to fix and this will become deprecated.

**Comment by ruflin** [comment]
* Is my understanding correct that if a user hits this, some changes might be missed so this would be a bug? 
* Does the Elasticsearch team have an issue to track this? Would be nice to link to it in the code based for any future engineer stumbling over it + having an issue on our end to track this change.
* Wanted to ask for an entry in the changelog but seems like we still don't have one :-( https://github.com/elastic/fleet-server/issues/602

**Comment by EricDavisX** [comment]
It looks ok to me for 7.15 release if you feel good.  Thanks for adding tests!  And ++ to editing comment to cite the ES side bug this works around so we can (at least periodically) check to see when /if this can be removed

**Comment by ruflin** [comment]
+1 on 7.15 as this might cause some issues if someone hits it so I consider it a bug.

**Review by blakerouse — APPROVED** [review]
Looks good, tested well. +1 for 7.15 as well.

**Comment by aleksmaus** [comment]
> Can you change comment to indicate when we expect ES to fix and this will become deprecated.

Reached out to elasticsearch team. The issue for "refresh" falls under phase 2 of https://github.com/elastic/elasticsearch/issues/71449

Once the phase is complete we can remove the refreshes from fleet-server.

**Comment by tbrooks8** [comment]
We're looking at phase 2 currently. We had a prototype (https://github.com/elastic/elasticsearch/pull/73134). However, we have slightly reconsidered the design based on work we plan on doing for multi-shard support. We would hope that the refresh work could go into 7.16, but we need to sync with the es-search team to ensure that our designs make sense.

**Pull request merged**